OpenSim/Simulation/Model/Model.cpp

```diff
@@ -1771,26 +1774,28 @@ Vector& Model::updControls(const SimTK::State &s) const
 
 void Model::markControlsAsValid(const SimTK::State& s) const
 {
+  std::unique_lock<std::mutex> lock(*controlsCacheLock);
     if( (!_system) || (!_modelControlsIndex.isValid()) ){
-        throw Exception("Model::markControlsAsValid() requires an initialized Model./n" 
+        throw Exception("Model::markControlsAsValid() requires an initialized Model./n"
             "Prior call to Model::initSystem() is required.");
     }
 
-    Measure_<Vector>::Result controlsCache = 
+    Measure_<Vector>::Result controlsCache =
         Measure_<Vector>::Result::getAs(_system->updDefaultSubsystem()
             .getMeasure(_modelControlsIndex));
     controlsCache.markAsValid(s);
 }
 
 void Model::setControls(const SimTK::State& s, const SimTK::Vector& controls) const
-{   
+{
+  std::unique_lock<std::mutex> lock(*controlsCacheLock);
```

Comment:
Indent 4 spaces. I don't know enough about locks but I don't see how this is used. Perhaps you'll have to explain to me in person.